libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -209,10 +218,18 @@ func getv2Path(c *configs.Cgroup) (string, error) {
 		return "", err
 	}
 
+	if rootless {
+		uid, err := detectUID()
+		if err != nil {
+			return "", err
+		}
+		slice = fmt.Sprintf("user.slice/user-%d.slice/user@%d.service/%s", uid, uid, slice)
```

Comment:
Am I missing something or this will end up with `user.slice` twice in the name?

Like, when `c.Parent` is not defined, this will end up as `user.slice/user-1000.slice/user@1000.service/user.slice`, no?

If the idea is to have a slice of the user manager for these containers (which is definitely a good idea!), maybe it should have a name other than `user.slice` to avoid confusion? Perhaps `containers.slice`? I do think `machine.slice` is not a terrible idea for this (also a good one for Rootfull containers.)

Reply:
Yes, this is intentional for crun compatibility.